project/src/com/google/daggerquery/protobuf/binding_graph.proto

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+syntax = "proto2";
+package dagerquery.models;
+
+import "com/google/daggerquery/protobuf/dependency.proto";
+
+option java_package = "com.google.daggerquery.protobuf";
+option java_outer_classname = "BindingGraphProto";
+
+message BindingGraph {
+
+  message ListWithDependencies {
+    repeated Dependency dependency = 1;
+  }
+
+  map<string, ListWithDependencies> adjacency_list = 1;
```

Comment:
The direction of the edges are actually important, 

```
class A {
  @Inject A(B b) { ... }
}
```

is different from

```
class B {
  @Inject B(A a) { ... }
}
```

You might need to adapt the data structure to ensure the relationship is preserved.

Could you please add an inline documentation to explain each class member (could be a short sentence, just so to avoid ambiguity of interpretation by the reader)? Thanks!

Reply:
I think this data structure provides directions 🤔

I mean with your example we will have adjacency_list[`A`] = [..., `B`, ...] for edge A->B (and `A` might not be presented in appropriate list for `B` if there is no edge B->A).

Yes, I will provide documentation in next commit, sure!

Reply:
Okay, thanks for the clarification! I was initially confused because the https://en.wikipedia.org/wiki/Adjacency_list page defined it for undirected graphs, but your definition would work as well. Please do add a comment (in the next commit is fine) to prevent people like me coming to bug you in the future :)